src/execution/experimental-executions.js

```diff
@@ -367,36 +527,30 @@ const createPoPMetric = (measure, mdObj, measureIndex) => {
                 format
             }
         },
-        sort: getMetricSort(get(measure, 'sort'), true),
+        sort: getMeasureSorting(popMeasure, mdObj),
         meta: {
             measureIndex,
             isPoP: true
         }
-    }];
-
-    if (generated) {
-        result.push(generated);
-    } else {
-        result.push(createPureMetric(measure, mdObj, measureIndex));
-    }
-
-    return result;
+    };
 };
 
-const createContributionPoPMetric = (measure, mdObj, measureIndex) => {
-    const date = getDate(mdObj);
+const createContributionPoPMetric = (popMeasure, mdObj, measureIndex, attributesMap) => {
+    const attributeUri = get(popMeasure, ['definition', 'popMeasureDefinition', 'popAttribute', 'uri']);
 
-    const generated = createContributionMetric(measure, mdObj, measureIndex);
-    const title = getPoPMetricTitle(get(measure, 'title'));
+    const originalMeasure = getOriginalMeasureForPoP(popMeasure, mdObj);
+
+    const generated = createContributionMetric(originalMeasure, mdObj, measureIndex, attributesMap);
+    const title = getBaseMetricTitle(get(popMeasure, 'title'));
 
     const format = CONTRIBUTION_METRIC_FORMAT;
     const hasher = partial(getGeneratedMetricHash, title, format);
 
-    const getMetricExpression = partial(getPoPExpression, date, `(${get(generated, 'definition.metricDefinition.expression')})`);
+    const getMetricExpression = partial(getPoPExpression, attributeUri, `(${get(generated, 'definition.metricDefinition.expression')})`);
```

Comment:
This is quite unreadable -> expand to more lines?

Reply:
fixed